### PR TITLE
Create and Add Lite service provider retreival service

### PR DESCRIPTION
**Pull request by Thumimku**
Public Issue: https://github.com/wso2/product-is/issues/16129

As explained in the Issue, we plan to introduce a method in the application mgt service which exposes a service provider object with elemental properties. Here the term “basic properties” is based on the issue's requirement. Hence service provider object contains,

    Basic application data from SP_APP table
        Service provider basic attributes
        Owner
        Claim configuration
        Local and Outbound configuration
    Claim mapping from SP_CLAIM_MAPPING table
    SP claim Dialect from SP_CLAIM_DIALECT table
    Service provider properties from SP_METADATA  table
    Role mapping from SP_ROLE_MAPPING table

Loaded service provider stores in a new cache for future reference.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+/**
+ * This class represents the basic service provider information.
+ * The purpose of this class is to store load service provider configuration with minimal load.
+ */
+public class LiteServiceProvider {
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java:25]: Would be good to have a better name..
Lite might give a wrong idea about this class :)

Comment (mefarazath) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java:25]: (This not actually the lite version of a service provider but a model we are using the extract a part of the application)

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    private ServiceProviderProperty[] spProperties = new ServiceProviderProperty[0];
+
+    private String applicationResourceId;
+
+    private String imageUrl;
+
+    private String accessUrl;
+
+    private boolean isDiscoverable;
+
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java:65]: How did we decide these are required among all attributes/data of an app?
What are the details we are leaving out?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java [hunk 3]
```diff
@@ -227,6 +228,32 @@ boolean doPostGetAllApplicationBasicInfo(ApplicationDAO appDAO, String tenantDom
...
+    /**
+     * Define any additional actions after getting a lite application excluding file based SPs.
+     *
+     * @param serviceProvider
+     * @param applicationName
+     * @param tenantDomain
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    boolean doPostGetLiteApplicationExcludingFileBasedSPs(LiteServiceProvider serviceProvider,
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java:252]: Do we really need these listener methods?

Comment (mefarazath) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java:252]: (practically)

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java (hunk 1 above)
Comment (chamathns) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LiteServiceProvider.java:25]: ServiceProviderMetadata?

**Comment by Thumimku** [comment]
Alternative solution provided via https://github.com/wso2/carbon-identity-framework/pull/4793

**Pull request closed without merging**